strictdoc/backend/sdoc/document_reference.py

```diff
@@ -1,9 +1,15 @@
+from __future__ import annotations
+
 from typing import Optional
+from typing import TYPE_CHECKING
 
-from strictdoc.backend.sdoc.models.document import Document
+if TYPE_CHECKING:
+    from strictdoc.backend.sdoc.models.document import Document
 
 
 class DocumentReference:
+    _document: Document | None
```

Comment:
The same as for the previous line: how changing this helps to achieve the goal of introducing the constants?

Reply:
By fixing the circular depency, PyCharm hinted to add a separate variable type hint. Not sure if this improves how python loads/initializes the classes.
Note: Found out that the used Union-Variant was [introduced with python 3.10](https://docs.python.org/3.10/library/typing.html#typing.Optional).
I will check if we can do without the hint or change it into Optional[Document].